discovery_engine_core/ai/ai/src/utils.rs

```diff
@@ -73,6 +75,20 @@ pub(crate) fn system_time_now() -> SystemTime {
     SystemTime::now()
 }
 
+/// Rank documents based on their score.
+///
+/// # Panics
+/// This function will panic if a score for a document is missing.
+pub fn rank<D, S>(documents: &mut [D], scores: &HashMap<D::Id, f32, S>)
+where
+    D: Document,
+    S: ::std::hash::BuildHasher,
```

Comment:
🎨 
```suggestion
    S: std::hash::BuildHasher,
```

or we could also add `hash::BuildHasher` to the `use` above and just use `BuildHasher` here

Reply:
I'm becoming less of a fan of polluting the global namespace with things that are super-specific and local.